Production/src/BaseTupleProducer.cc

```diff
@@ -11,6 +11,32 @@
 #include "DataFormats/L1Trigger/interface/BXVector.h"
 #include "h-tautau/Core/include/DiscriminatorIdResults.h"
 
+
+int TupleStore::tuple_counter = 0;
+
+std::unique_ptr<ntuple::EventTuple> TupleStore::eventTuple_ptr;
+
+ntuple::EventTuple& TupleStore::GetTuple()
+{
+  ROOT::CompressionSettings(ROOT::kLZ4, 5);
```

Comment:
- the output file could already be opened, need to set compression settings to the file it self.
- should be done inside if